src/sharding/ShardingManager.js

```diff
@@ -101,6 +107,36 @@ class ShardingManager extends EventEmitter {
     return shard;
   }
 
+  /**
+   * Kills all Shards without them respawning.
+   */
+  killAll() {
+    const _respawn = this.respawn;
+    this.respawn = false;
+    for (const shard of this.shards.values()) shard.kill();
+    this.respawn = _respawn;
+  }
+
+  /**
+   * Checks the Shard amount, then respawns the correct amount of Shards.
+   * @param {number} [delay=5500] How long to wait in between spawning each shard (in milliseconds)
+   * @param {boolean} [waitForReady=true] Whether to wait for a shard to become ready before continuing to another
+   * @returns {Promise<Collection<number, Shard>>}
+   */
+  async rebalanceShards(delay = 5500, waitForReady = true) {
+    // Obtain/verify that the totalShards is set to 'auto' for shards to spawn
+    if (!this.autoShards) throw new TypeError('CLIENT_INVALID_OPTION', 'Set totalShards', 'to auto.');
+    this.totalShards = await Util.fetchRecommendedShards(this.token);
```

Comment:
This seems redundant, just directly pass `'total'` to `this.respawn` a few lines below.

Reply:
Ok, i'm changing it.